docs/visualizing-your-sql-queries/plotting-with-seaborn.md

```diff
@@ -1,14 +1,559 @@
+
+A few examples denoting this distinction are shown below.
+
+### Axes-level
+
+Suppose we want to identify whether gentrification increases the average salary of two regions, using the data we downloaded above. We first save our [CTE (Common Table Expression)](https://ploomber-sql.readthedocs.io/en/latest/advanced-querying-techniques/ctes) named `levels_example` that takes in the columns, `average_salary`, `ratio_of_urban_inhabitants`, and `region`, and filters for two regions, 'central Bohemia' and 'east Bohemia', from the `s1.district` table.
+
+```{code-cell} ipython3
+%%sql --save levels_example
+SELECT average_salary, ratio_of_urban_inhabitants, region
+FROM s1.district
+WHERE region IN ('central Bohemia', 'east Bohemia');
+```
+
+The result of the CTE is saved as a pandas `DataFrame()`:
+
+```{code-cell} ipython3
+result = %sql SELECT * FROM levels_example
+df = result.DataFrame()
+```
+
+You can determine what is returned using Python's `type()` function:
+
+```{code-cell} ipython3
+plt.rcParams["figure.dpi"] = 300  # high resolution
+
+scatter_plt = sns.scatterplot(
+    data=df, x="ratio_of_urban_inhabitants", y="average_salary", hue="region"
+)
+print(type(scatter_plt))
+```
+
+Notice that the `sns.scatterplot()` function returns a Matplotlib `Axes` object, a single plot that is inclusive of the data from both regions. Other seaborn functions, including `regplot`, `boxplot`, `kdeplot`, and many others, also return Matplotlib `Axes` objects. Therefore, we can use various Matplotlib axes [commands](https://seaborn.pydata.org/generated/seaborn.axes_style.html) to modify the Seaborn figure.
+
+### Figure-level
+
+On the other hand, we can show that `sns.relplot()` returns a `FacetGrid` object, creating separate plots for the regions:
+
+```{code-cell} ipython3
+facet_plt = sns.relplot(
+    data=df, x="ratio_of_urban_inhabitants", y="average_salary", col="region"
+)
+print(type(facet_plt))
+```
+
+Other Figure-level seaborn functions include `catplot`, `displot`, `pairplot`, and `jointplot`.
+
+```{important}
+The legends are placed outside the plot if a Figure-level plotting function is used. See scatterplot section below.
+```
+
+Let's now jump into one of the most simple, yet essential, data visualizations: the bar plot.
+
+## Barplots
+
+The most basic [`seaborn.barplot()`](https://seaborn.pydata.org/generated/seaborn.barplot.html) function takes in a categorical and a numeric variable as <b>encodings</b>. A second layer of grouping, preferably with another categorical variable, can be added with the `hue` argument.
```

Comment:
Takes in a -> takes a

Please run this via a grammar check

Reply:
With the hue argument -> to the hue argument

Reply:
resolved, good to close